### PR TITLE
update README

### PR DESCRIPTION
**Pull request by IliaOzhmegov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,23 +200,33 @@ Initially we intended to use Bitcoin data for our research. But for reasons abov
...
+
+**Sequences**
+
+We have created a shortened version of the raw data - the *"SP500"* csv file. Since more reliable data is from aproximatelly 1962, we have discarted the data before January 2, 1962. 
+
+Next, as part of the data preparation, we have create the sequences of these data. They were created by building a custom sequence distributor (see *sequence_distributor.py*) to slice data into the small windows of 30 days each, which will be used as predictors, and the slices with the span of 5 days - the target values that need to be predicted. The predictor and target data were scaled separetelly using different approaches. For  example, the 30-day predictor sequences were scaled by taking the minimum and maximus values: 4.5% and 95.5% respectively. So overall we have created 14,832 sequences. These sequences were used as an inputs for our Time Series Neural Networks: TDNN, LSTM, RNN.
```
Comment (IliaOzhmegov) [inline README.md:223]: I'd change:
`For  example, the 30-day predictor sequences were scaled by taking the minimum and maximus values: 4.5% and 95.5% respectively`
to 
`For  example, the 30-day predictor sequences were scaled by taking the minimum and **maximum** values as 0.04(45) and 0.95(45) respectively. The choice of such values can be explained by scale factor of plot function`

File: README.md [hunk 2]
```diff
@@ -228,22 +238,56 @@ Speaking of analyzing the raw visual data,  we had to create a certain pipeline
...
 5. As we get our data converted into convenient **HLOC** format we start the forecasting process using our ANN models by inputting those sequences into them. As an output we get the **HLOC** and **t<sub>k+1</sub>** values of the predicted candles. These predicted and input values together can be used by our candle plot generator to obtain the whole candle plot with forecasted candles. 
 
+
+
 **Candles.**
 
-Other part of our data preparation process we have created an images of the candles. This was a tedious process, where we had to create candle plots and "cut-out" each candle separately. In addition, we have added a background noise. Here are the samples of these candles.
+Other part of our data preparation process we have created an images of the candles (see *candle_builder.py*). This was a tedious process, where we had to create candle plots and "cut-out" each candle separately. We did so by drawing the 30-day slices of the data, then slicing each of them into individual plots. These individual plots contain a sindle candle with preserved spatial properties (including the center). The size of each candle box is 34 (width) x 200 (hight) pixels.
```
Comment (IliaOzhmegov) [inline README.md:245]: sindle

File: README.md [hunk 3]
```diff
@@ -228,22 +238,56 @@ Speaking of analyzing the raw visual data,  we had to create a certain pipeline
...
 
+
+
 **Candles.**
 
-Other part of our data preparation process we have created an images of the candles. This was a tedious process, where we had to create candle plots and "cut-out" each candle separately. In addition, we have added a background noise. Here are the samples of these candles.
+Other part of our data preparation process we have created an images of the candles (see *candle_builder.py*). This was a tedious process, where we had to create candle plots and "cut-out" each candle separately. We did so by drawing the 30-day slices of the data, then slicing each of them into individual plots. These individual plots contain a sindle candle with preserved spatial properties (including the center). The size of each candle box is 34 (width) x 200 (hight) pixels. 
+
+For model robustness, we have introduced a parameter *lambda λ* that controls the transparency of the candle color. In addition, we have added some background noise. Here are the samples of these candles.
```
Comment (IliaOzhmegov) [inline README.md:247]: alpha - transparency, but also width of the each plots and slight swing to left and right

File: README.md [hunk 4]
```diff
@@ -228,22 +238,56 @@ Speaking of analyzing the raw visual data,  we had to create a certain pipeline
...
+
 
 
+<img src="plots/candles/2_0.png" alt="ACF_plot" style="zoom:100%;" />      <img src="plots/candles/3_26.png" alt="ACF_plot" style="zoom:100%;" />       <img src="plots/candles/3_20.png" alt="ACF_plot" style="zoom:100%;" />        <img src="plots/candles/0_3.png" alt="ACF_plot" style="zoom:100%;" />       <img src="plots/candles/0_1.png" alt="ACF_plot" style="zoom:100%;" />
 
-<img src="plots/candles/2_0.png" alt="ACF_plot" style="zoom:100%;" />      <img src="plots/candles/3_26.png" alt="ACF_plot" style="zoom:100%;" />       <img src="plots/candles/3_20.png" alt="ACF_plot" style="zoom:100%;" />        <img src="plots/candles/4_2.png" alt="ACF_plot" style="zoom:100%;" />       <img src="plots/candles/232_22.png" alt="ACF_plot" style="zoom:100%;" />
 
 
+The names of the images of the individual candles are not random. They folow a pre-defined convention where first number of the image name means it belongs to the according sequence file, and the second number means the corresponding row number of that particular sequence. For example, if we are looking at the candle with name "7_1.png", it means that it belongs to the 7th sequence, 1st row in that 7th sequence.
```
Comment (IliaOzhmegov) [inline README.md:255]: 1 means the 2nd row of the sequence file

File: README.md [hunk 5]
```diff
@@ -228,22 +238,56 @@ Speaking of analyzing the raw visual data,  we had to create a certain pipeline
...
 
 
->>>>>>> master
 
-The data we are using is inclined towards a bullish market. 
+**Selector.**
+
+Our initial idea was to create a *selector* that would do a sophisticated slicing of the candle plots, however due to time constrains it is not ready. So the way it would work - it would take a part of the image with size 1150 x 210 pixels, and similarly to the *candle_builder*, would cut-off the slices of the image with candles with step size of 3 pixels in between the cuts. 
+
+it would also calculate the gradient on x-axis that corresponds to and shows some edges of the candle. It would determine the center of the image and would add 17 pixels on each side of the of the slice. It would result in some brighter clusters and combine them together.
```
Comment (IliaOzhmegov) [inline README.md:265]: **Please replace you paragraph with** 
However, further we are planning to change the naive approach to a more robust one, which will calculate the x of "the center of mass" of every candle and then slice 17 pixels margin in both directions from the center.  The center of mass can be calculated with OpenCV library.

File: README.md [hunk 6]
```diff
@@ -228,22 +238,56 @@ Speaking of analyzing the raw visual data,  we had to create a certain pipeline
...
+*"check_iterpretator.py"* - checks how our interpretator works. 
+
+
+
+
+
+
+>>>>>>> master
+>>>>>>>
+>>>>>>>
```
Comment (IliaOzhmegov) [inline README.md:290]: Please, remove it, it looks like I forgot

Comment (Len-X) [reply README.md:290]: I already did

File: README.md (hunk 1 above)
Comment (Len-X) [reply README.md:223]: done

File: README.md (hunk 2 above)
Comment (Len-X) [reply README.md:245]: sorry, did't get this part. what is sindle? :)

File: README.md (hunk 3 above)
Comment (Len-X) [reply README.md:247]: done

File: README.md (hunk 4 above)
Comment (Len-X) [reply README.md:255]: 1 means 2nd row?

okay. updated.

File: README.md (hunk 5 above)
Comment (Len-X) [reply README.md:265]: done

**Pull request merged**